tools/upload_text.php

```diff
@@ -269,32 +269,48 @@
         // replace filename
         $zipext = ".zip";
         $name = $projectid.$indicator.$zipext;
-        $path_to_file = "$projects_dir/$projectid/";
-        $location = $path_to_file.$name;
+        $location = "$project->dir/$name";
         ensure_path_is_unused( $location );
         copy($uploaded_file, $location);
         $have_file = TRUE;
         if ($stage == 'smooth_avail')
         {
-            $smooth_dir = "$projects_dir/$projectid/smooth/";
-            // make smooth folder if not exists
-            if(!is_dir($smooth_dir))
+            $project->delete_smoothreading_dir();
+            $smooth_dir = "$project->dir/smooth";
+            if(!mkdir($smooth_dir))
             {
-                if(!mkdir($smooth_dir))
-                {
-                    die("Could not create smooth directory");
-                }
-            }
-            else
-            {
-                delete_contents($smooth_dir);
+                die("Could not create smooth directory");
             }
             // unzip into smooth folder
             $zip = new ZipArchive;
             if ($zip->open($uploaded_file))
             {
                 $zip->extractTo($smooth_dir);
                 $zip->close();
+                // if there is a html or htm file zip it (with images) for download
+                $html_files = glob("$smooth_dir/*.{htm,html}", GLOB_BRACE);
+                if($html_files)
+                {
+                    // assume only one html file
+                    $file = $html_files[0];
+                    $zip = new ZipArchive;
```

Comment:
You might consider using `create_zip_from()` in `misc.inc`. That will remove all the paths and have everything within the zip in a single directory, which may or may not be what you want.